test/src/unittests/tonal/test_audio2midi.py

```diff
@@ -0,0 +1,312 @@
+            midi_buffer_duration=midi_buffer_duration,
+            min_note_change_period=min_note_change_period,
+            max_frequency=max_frequency,
+            min_frequency=min_frequency,
+            n_notes_tolerance=n_notes_tolerance,
+            onset_tolerance=onset_tolerance,
+            midi_note_tolerance=midi_note_tolerance,
+        )
+
+    def testSeparatedNotes(self):
+        frame_size = 8192
+        sample_rate = 44100
+        hop_size = 16
+        loudness_threshold = -50
+        pitch_confidence_threshold = 0.8
+        min_frequency = 103.83
+        max_frequency = 659.26
+        midi_buffer_duration = 0.05
+        min_note_change_period = 0.02
+        min_onset_period = 0.015
+        min_offset_period = 0.015
+        min_occurrence_rate = 0.05
+        n_notes_tolerance = 0
+        onset_tolerance = 0.01
+        midi_note_tolerance = 0
+
+        stem = "387517__deleted_user_7267864__saxophone-going-up"
+        audio_path = Path("recorded") / f"{stem}.wav"
+        reference_path = Path("audio2midi") / f"{stem}.npy"
+
+        self.runARealCase(
+            audio_path=audio_path,
+            reference_path=reference_path,
+            sample_rate=sample_rate,
+            frame_size=frame_size,
+            hop_size=hop_size,
+            pitch_confidence_threshold=pitch_confidence_threshold,
+            loudness_threshold=loudness_threshold,
+            max_frequency=max_frequency,
+            min_frequency=min_frequency,
+            midi_buffer_duration=midi_buffer_duration,
+            min_note_change_period=min_note_change_period,
+            min_onset_period=min_onset_period,
+            min_offset_period=min_offset_period,
+            min_occurrence_rate=min_occurrence_rate,
+            n_notes_tolerance=n_notes_tolerance,
+            onset_tolerance=onset_tolerance,
+            midi_note_tolerance=midi_note_tolerance,
+        )
+
+    def runARealCase(
+        self,
+        audio_path: str,
+        reference_path: str,
+        sample_rate: int,
+        frame_size: int,
+        hop_size: int,
+        pitch_confidence_threshold: float,
+        loudness_threshold: float,
+        max_frequency: float,
+        min_frequency: float,
+        midi_buffer_duration: float,
+        min_note_change_period: float,
+        min_onset_period: float = 0.075,
+        min_offset_period: float = 0.2,
+        min_occurrence_rate: float = 0.5,
+        n_notes_tolerance: int = 0,
+        onset_tolerance: float = 0.01,
+        offset_tolerance: float = 0.05,
+        midi_note_tolerance: int = 0,
+    ):
+        filename = join(testdata.audio_dir, audio_path)
+        if sys.platform == "darwin":
+            import soundfile as sf
+
+            audio, _ = sf.read(filename, dtype="float32")
+            if audio.ndim > 1:
+                audio = audio[:, 0]
+        else:
+            audio = MonoLoader(filename=filename, sampleRate=sample_rate)()
```

Comment:
Use `MonoLoader` for both cases for consistency and also to avoid additional dependency.

Reply:
Right, but I running tests in OSX. So, I need it to run them. Once we resolve all comments I might remove it. Let's fix the other parts and I will do it.